controllers/cloudinit/controlplane_init.go

```diff
@@ -101,7 +103,7 @@ func NewInitControlPlane(input *ControlPlaneInitInput) (*CloudConfig, error) {
 	cloudConfig.RunCommands = append(cloudConfig.RunCommands,
 		"set -x",
 		scriptPath(disableHostServicesScript),
-		fmt.Sprintf("%s %d.%d", scriptPath(installMicroK8sScript), kubernetesVersion.Major(), kubernetesVersion.Minor()),
+		fmt.Sprintf("%s %d.%d %s", scriptPath(installMicroK8sScript), kubernetesVersion.Major(), kubernetesVersion.Minor(), input.Confinement),
```

Comment:
I would make this a `%q` to ensure that the bash script gets an (empty) argument even when confinement is not strict.

```suggestion
		fmt.Sprintf("%s %d.%d %q", scriptPath(installMicroK8sScript), kubernetesVersion.Major(), kubernetesVersion.Minor(), input.Confinement),
```